### DISCUSSION TITLE
Problems with C# Native module in 0.73 version

### DISCUSSION THREAD

**Discussion by jCrCaT** [microsoft/react-native-windows#12918] [Q&A] [answered]
Hello,

I've been perusing the react-native-windows documentation and noticed that most examples are tailored for version 0.62. However, while attempting to develop my own native module in version 0.73, I've encountered a significant hurdle—the C# environment fails to recognize any of the specified using namespaces or annotations. For instance, the following snippet of code:

`using System;
using Microsoft.ReactNative.Managed;

namespace NativeModuleSample
{
    [ReactModule]
    class FancyMath
    {
        [ReactMethod("add")]
        public double Add(double a, double b)
        {
            double result = a + b;
            return result;
        }
    }
}`

This code runs into issues in the 0.73 environment, with none of the annotations like [ReactModule] being recognized. Conversely, when I switch to a react-native-windows 0.63 setup, the annotations and namespaces are fully recognized without any problems.

It leads me to suspect that there might have been changes in the approach to building native modules in subsequent releases of react-native-windows. Could you confirm whether there have been any updates to the process for creating native modules in the latest versions? Furthermore, is there an updated set of documentation on the horizon to guide developers through these changes?

Thank you for your assistance!!

**Comment by chrisglein** [comment, marked as answer]
> It leads me to suspect that there might have been changes in the approach to building native modules in subsequent releases of react-native-windows. Could you confirm whether there have been any updates to the process for creating native modules in the latest versions?

Yes, confirmed. Between 0.63 and later releases there were changes in the native module authorship. 

> Furthermore, is there an updated set of documentation on the horizon to guide developers through these changes?

I recommend you get started here: https://microsoft.github.io/react-native-windows/docs/native-modules
And there are specifically C# examples here: https://github.com/microsoft/react-native-windows-samples/tree/main/samples/NativeModuleSample/csharp

Hope that helps!

**Reply by jCrCaT** [reply]
Upon reviewing my code, I noticed it was written in C++. Therefore, when installing react-native-windows, it's important to use the language parameter: --language cs 

:)

Thank you!!
